From: Dermot Bradley <dermot_bradley@yahoo.com>
Date: Thu, 10 Dec 2020 23:52 +0000
Subject: ensure that small filesystems are Y2038 ok

When formatting a partition with a EXT2/3/4 filesystem, if the partition
specified is 512M or smaller in size then the "small" entry from
/etc/mke2fs.conf is used, in addition to the defaults in that file, for
the new filesystem's settings. Currently the "small" entry specifies an
inode size of 128bytes which is insufficient to hold timestamps beyond
2038.

After booting a system with an affected EXT2/3/4 filesystem the kernel
buffer will show a warning:

$ dmesg | grep 2038

  filesystem being remounted at / supports timestamps until 2038

This patch removes the 128 byte inode size entry from the 'small'
definition and so the formatting tool will fallback to the default
value of 256 byte inodes which will ensure no Y2038 issues. The disk
space overhead when formatting after this change will slightly increase.

---

diff -aur a/misc/mke2fs.conf.in b/misc/mke2fs.conf.in
--- a/misc/mke2fs.conf.in
+++ b/misc/mke2fs.conf.in
@@ -16,7 +16,6 @@
 	}
 	small = {
 		blocksize = 1024
-		inode_size = 128
 		inode_ratio = 4096
 	}
 	floppy = {
